hibernate-core/src/test/java/org/hibernate/orm/test/batchfetch/StreamTest.java

```diff
@@ -0,0 +1,120 @@
+package org.hibernate.orm.test.batchfetch;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.query.spi.QueryImplementor;
+
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.JiraKey;
+import org.hibernate.testing.orm.junit.ServiceRegistry;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.Setting;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@DomainModel(
+		annotatedClasses = { StreamTest.Child.class, StreamTest.Parent.class }
+)
+@SessionFactory
+@ServiceRegistry(settings = @Setting(name = AvailableSettings.DEFAULT_BATCH_FETCH_SIZE, value = "2"))
+@JiraKey("HHH-15449")
+public class StreamTest {
+
+	public static final String FIELD_VALUE = "afield";
+
+	@BeforeEach
+	public void setUp(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					Parent parent = new Parent( FIELD_VALUE );
+					session.persist( parent );
+					session.persist( new Child( parent ) );
+				}
+		);
+	}
+
+	@Test
+	public void testGetResultStream(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					QueryImplementor<Child> query = session
+							.createQuery( "select c from Child as c where c.parent.someField=:someField", Child.class )
+							.setParameter( "someField", FIELD_VALUE );
+					Stream<Child> resultStream = query.getResultStream();
+					List<Child> children = resultStream.collect( Collectors.toList() );
```

Comment:
maybe test directly on stream rather than doing a collect?  if a collect can trigger a close or other event, then it may behave differently than when processing individual stream entries(?)

Reply:
Streams are not implicitly closed, so this is fine AFAICT as the `Stream#close` method and hence `endLoading` are not called.